src/components/MarkersDisplay/Annotations/AnnotationSetSelect.js

```diff
@@ -121,88 +131,219 @@ const AnnotationSetSelect = ({
-    annotationSet.items = items;
-    setSelectedAnnotationSets((prev) => [...prev, annotationSet.label]);
-    setDisplayedAnnotationSets((prev) => [...prev, annotationSet]);
+  const handleDropdownKeyPress = (e) => {
+    // Close the dropdown on 'Escape' keypress if it is open
+    if (e.key === 'Escape') {
+      e.preventDefault();
+      if (isOpenRef.current) toggleDropdown();
+    }
+
+    // Toggle dropdown on Enter/Space keypresses
+    if (e.key === 'Enter' || e.key === ' ') {
+      e.preventDefault();
+      toggleDropdown();
+    }
+    // Open the dropdown and move focus to first option on ArrowDown/ArrowUp keypresses
+    if (e.key === 'ArrowDown' || e.key === 'ArrowUp') {
+      if (!isOpenRef.current) {
+        setIsOpen(true);
+        // Keep the container scrolled to top. Without this the first option 
+        // gets out of view when dropdown is programatically opened
+        setTimeout(() => {
+          if (dropDownRef.current) dropDownRef.current.scrollTop = 0;
+        }, 0);
+      }
+
+      // Move focus to the first option in the list
+      const firstOption = document.querySelector(".annotations-dropdown-item");
+      if (firstOption) {
+        firstOption.focus();
+        setCurrentIndex(0);
+      }
+    }
+  };
+
+  /**
+   * Handle keyboard events for each annotation set option.
+   * @param {Event} e keyboard event
+   */
+  const handleAnnotationSetKeyPress = (e) => {
+    const allOptions = dropDownRef.current.children;
+    let nextIndex = currentIndex.current;
+    switch (e.key) {
+      case 'Enter':
+      case ' ':
+        // On Enter/Space select the focused annotation set
+        e.preventDefault();
+        const option = timedAnnotationSets.filter((a) => e.target.id == a.label)[0];
+        if (option != undefined) {
+          handleSelect(option);
+        } else {
+          handleSelectAll(e);
+        }
+        break;
+      case 'ArrowDown':
+        e.preventDefault();
+        // Move to next option on ArrowDown keypress and wraps to first option when end is reached
+        nextIndex = (currentIndex.current + 1) % allOptions.length;
+        break;
+      case 'ArrowUp':
+        e.preventDefault();
+        // Move to previous option on ArrowUp keypress and wraps to last option when top is reached
+        nextIndex = (currentIndex.current - 1 + allOptions.length) % allOptions.length;
+        break;
+      case 'Escape':
+        e.preventDefault();
+        // Close the dropdown and move focus to dropdown button
+        toggleDropdown();
+        selectButtonRef.current.focus();
+        break;
+      case 'Tab':
+        // Close dropdown and move focus out to the next element in the DOM
+        toggleDropdown();
+        break;
+    }
+
+    // Update focus on the selected option using ArrowDown/ArrowUp keys
+    if (nextIndex !== currentIndex.current) {
+      allOptions[nextIndex].focus();
+      setCurrentIndex(nextIndex);
+    }
```

Comment:
Not a big deal but if an option in the dropdown is only partially rendered due to scrolling then when focus is given the option isn't scrolled fully into view.  The [select-only combobox example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/) ARIA pattern scrolls in this scenario.

Reply:
It seems I can't make it work with `aria-activedescendant` attribute, and I can't figure out why. It could be clashing with something else we are doing in the code?
However I was able to call `scrollIntoView()` on the focused option to scroll into view programmatically.